database-plugins/src/main/java/co/cask/hydrator/plugin/db/batch/source/DBSource.java

```diff
@@ -66,6 +83,127 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     dbManager.validateJDBCPluginPipeline(pipelineConfigurer, getJDBCPluginId());
   }
 
+  class ListTablesRequest {
+
+    @Description("JDBC connection string including database name.")
+    public String connectionString;
+
+    @Description("User to use to connect to the specified database. Required for databases that " +
+      "need authentication. Optional for databases that do not require authentication.")
+    @Nullable
+    public String user;
+
+    @Description("Password to use to connect to the specified database. Required for databases that " +
+      "need authentication. Optional for databases that do not require authentication.")
+    @Nullable
+    public String password;
+
+    @Description("Name of the JDBC plugin to use. This is the value of the 'name' key defined in the JSON file " +
+      "for the JDBC plugin.")
+    public String jdbcPluginName;
+
+    @Description("Type of the JDBC plugin to use. This is the value of the 'type' key defined in the JSON file " +
+      "for the JDBC plugin. Defaults to 'jdbc'.")
+    @Nullable
+    public String jdbcPluginType = "jdbc";
+  }
+
+  class GetSchemaRequest extends ListTablesRequest {
+    @Description("Select query for which schema has to be populated.")
+    public String importQuery;
```

Comment:
I think we should rename this.  We don't want people to confuse this with the importQuery for the source, as this should be as lightweight of a query as possible.